qiskit/circuit/quantumcircuit.py

```diff
@@ -1533,6 +1534,381 @@ def _rebind_definition(self, instruction, parameter, value):
                             op.params[idx] = param.bind({parameter: value})
                         self._rebind_definition(op, parameter, value)
 
+    def barrier(self, *qargs):
+        """Apply :class:`~qiskit.circuit.Barrier`. If qargs is None, applies to all."""
+        from .barrier import Barrier
+        qubits = []
+
+        if not qargs:  # None
+            for qreg in self.qregs:
+                for j in range(qreg.size):
+                    qubits.append(qreg[j])
+
+        for qarg in qargs:
+            if isinstance(qarg, QuantumRegister):
+                qubits.extend([qarg[j] for j in range(qarg.size)])
+            elif isinstance(qarg, list):
+                qubits.extend(qarg)
+            elif isinstance(qarg, range):
+                qubits.extend(list(qarg))
+            elif isinstance(qarg, slice):
+                qubits.extend(self.qubits[qarg])
+            else:
+                qubits.append(qarg)
+
+        return self.append(Barrier(len(qubits)), qubits, [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def h(self, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.HGate`."""
+        from .library.standard_gates.h import HGate
+        return self.append(HGate(), [qubit], [])
```

Comment:
i think we should use compose here to get that behaviour going. @ajavadia and @Cryoris what do you think

Reply:
compose doesn't work for gates now, only circuits. it should i'll extend it later.